mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -1594,25 +1644,34 @@ async fn electrum_request_multi(
     }
     drop(connections);
     if futures.is_empty() {
-        return ERR!("All electrums are currently disconnected");
+        return Err(JsonRpcError::new(
+            client.coin_name(),
+            &request,
+            JsonRpcErrorType::Transport("All electrums are currently disconnected".to_string()),
+        ));
     }
 
-    match request {
+    match &request {
         JsonRpcRequestEnum::Single(single) if single.method == "server.ping" => {
             // server.ping must be sent to all servers to keep all connections alive
             return select_ok(futures)
                 .map(|(result, _)| result)
-                .map_err(|e| ERRL!("{:?}", e))
+                .map_err(|e| {
+                    JsonRpcError::new(client.coin_name(), &request, JsonRpcErrorType::Transport(e.to_string()))
+                })
                 .compat()
                 .await;
         },
         _ => (),
     }
 
-    let (res, no_of_failed_requests) = select_ok_sequential(futures)
-        .compat()
-        .await
-        .map_err(|e| ERRL!("{:?}", e))?;
+    let (res, no_of_failed_requests) = select_ok_sequential(futures).compat().await.map_err(|e| {
+        JsonRpcError::new(
+            client.coin_name(),
+            &request,
+            JsonRpcErrorType::Transport(format!("{:?}", e)),
+        )
+    })?;
```

Comment:
Well, this is really tricky to handle since `select_ok_sequential` tries to get a result by trying one electrum at a time from the electrums list and the errors are returned in a vec `Vec<JsonRpcError>`.

I think to fix this `electrum_request` https://github.com/KomodoPlatform/atomicDEX-API/blob/16579c69f2458ba6428461997ab49c117a1b08d1/mm2src/coins/utxo/rpc_clients.rs#L2778 should return only `JsonRpcErrorType::Transport` which means moving this https://github.com/KomodoPlatform/atomicDEX-API/blob/16579c69f2458ba6428461997ab49c117a1b08d1/mm2src/coins/utxo/rpc_clients.rs#L2788-L2797 outside the `electrum_request` function.
This is not ideal, but it's better than matching the errors in the vector and returning the `JsonRpcErrorType::InvalidRequest` if there is any, otherwise returning `JsonRpcErrorType::Transport`. What do you think @sergeyboyko0791 @borngraced ?

Reply:
agree with you.